h2o-algos/src/main/java/hex/tree/gbm/GBM.java

```diff
@@ -916,6 +917,19 @@ private void rollupMinMaxPreds(DTree tree, DTree.Node node, float[] mins, int mi
       return new GBMModel(modelKey, parms, new GBMModel.GBMOutput(GBM.this));
     }
 
+    @Override
+    protected void doInTrainingCheckpoint() {
+      try {
+        String modelFile = _parms._in_training_checkpoints_dir + "/" + _model._key.toString() + "." + _model._output._ntrees;
+        GBMModel modelClone = _model.clone();
+        modelClone.setInputParms(_parms);
+        modelClone._key = Key.make(_model._key + "." +  _model._output._ntrees);
+        modelClone._output.changeModelMetricsKey(modelClone._key);
```

Comment:
is `_output` also automatically cloned by default when cloning the model?

Reply:
Good point, its not, but it won't affect the final model. I will clone it also to have it clean, thanks